### PR TITLE
created model classes, associations and data processing

### PR DESCRIPTION
**Pull request by xtrahuman**
- [x] Create models
  - Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
  - Set up associations between models.
- [x] use models to insert data in rails console
- [x] Create custom methods
- my models include this additional methods.
    - Users
         - A method that returns the 3 most recent posts for a given user.
    - Posts
         - A method that updates the posts counter for a user.
         - A method which returns the 5 most recent comments for a given post.
    - Comments
         - A method that updates the comments counter for a post.
    - Likes
         - A method that updates the likes counter for a post.

- Additionally, made current user data available to all controllers.

   - Created a method called current_user in my ApplicationController.
   - A current_user method returns the first user from a database.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :user
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
+
+  def last_five_posts
+    posts.order(created_at: :desc).limit(5)
+  end
```
Comment (Dipeshtwis) [inline app/models/post.rb]: - [ ] I think you have to show 5 recent comments and not posts here.

Hint:
```ruby
def recent_comments
  comments.order('created_at Desc').limit(5)
end
 ```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :user
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
```
Comment (Dipeshtwis) [inline app/models/post.rb]: [Optional]

- I highly suggest that you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private. Similarly, you can do this with `update_comments_counter` and `update_likes_counter` as well.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :user
+
```
Comment (Dipeshtwis) [inline app/models/post.rb]: - [ ] Also, you have not associated the `post` model with `likes` and `comments`. Please add association here between `post and comment` and `post and likes`


```ruby
class Post < ApplicationRecord
 ..
 has_many :comments
 has_many :likes
end
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @xtrahuman, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @xtrahuman,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**